internal/host/installcmd.go

```diff
@@ -179,3 +189,36 @@ func getBootDevice(log logrus.FieldLogger, hwValidator hardware.Validator, host
 func GetDeviceFullName(name string) string {
 	return fmt.Sprintf("/dev/%s", name)
 }
+
+func getMCOImageFromOpenshiftRelease(releaseImage string, pullSecret string) (string, error) {
+	// write pull secret to a temp file
+	ps, err := ioutil.TempFile("", "registry-config")
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		ps.Close()
+		os.Remove(ps.Name())
+	}()
+	_, err = ps.Write([]byte(pullSecret))
+	if err != nil {
+		return "", err
+	}
+	// flush the buffer to ensure the file can be read
+	ps.Close()
+
+	// get mco image from release image
+	cmd := exec.Command("oc", "adm", "release", "info",
```

Comment:
is this command relevant for on prem?

Reply:
This is to remove the mco image hardcoded in assisted-installer and get it from release image instead. That is for both cloud and onprem usecases. For disconnected onprem, we've agreed to get the image from an environment variable

Reply:
but here i don't see any condition that don't run this command for on-prem or did i missed something?

Reply:
It's a work in progress. There will be a condition for disconnected onprem.